### PR TITLE
DB: Database connection setup

### PR DESCRIPTION
**Pull request by MHA10**
- Database connection setup
- Postgres
- Docker
- TypeORM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,73 +1,154 @@
...
-    <a href="https://opencollective.com/nest#sponsor"  target="_blank"><img src="https://img.shields.io/badge/Support%20us-Open%20Collective-41B883.svg" alt="Support us"></a>
-  <a href="https://twitter.com/nestframework" target="_blank"><img src="https://img.shields.io/twitter/follow/nestframework.svg?style=social&label=Follow"></a>
-</p>
-  <!--[![Backers on Open Collective](https://opencollective.com/nest/backers/badge.svg)](https://opencollective.com/nest#backer)
-  [![Sponsors on Open Collective](https://opencollective.com/nest/sponsors/badge.svg)](https://opencollective.com/nest#sponsor)-->
+Install the dependencies:
 
-## Description
+```bash
+npm
```
Comment (MHassanTariq) [inline README.md]: npm _install_ to install the dependencies?

File: README.md [hunk 2]
```diff
@@ -1,73 +1,154 @@
...
+### Generate a migration from existing table schema <br />
 
-## Installation
+Automatic migration generation creates a new migration file and writes all sql queries that must be executed to update the database
+
+Generate migration locally
 
 ```bash
-$ yarn install
+TableName=<name> npm run local-migration:generate
```
Comment (MHassanTariq) [inline README.md]: Are you sure that `TableName=<name>` will come in the start of the command?

File: package.json [hunk 3]
```diff
@@ -12,6 +12,22 @@
...
+    "local-migration:show": "NODE_ENV=dev ts-node ./node_modules/typeorm/cli.js migration:show -d ./src/config/datasource.ts",
+    "prod-migration:show": "NODE_ENV=production ts-node ./node_modules/typeorm/cli.js migration:show -d ./dist/config/datasource.js",
+    "local-migration:schema-sync": "NODE_ENV=dev ts-node ./node_modules/typeorm/cli.js schema:sync -d ./src/config/datasource.ts",
+    "prod-migration:schema-sync": "NODE_ENV=production ts-node ./node_modules/typeorm/cli.js schema:sync -d ./dist/config/datasource.js",
+    "local-migration:schema-log": "NODE_ENV=dev ts-node ./node_modules/typeorm/cli.js schema:log -d ./src/config/datasource.ts",
+    "prod-migration:schema-log": "NODE_ENV=production ts-node ./node_modules/typeorm/cli.js schema:log -d ./dist/config/datasource.js",
+    "local-migration:schema-drop": "NODE_ENV=dev ts-node ./node_modules/typeorm/cli.js schema:drop -d ./src/config/datasource.ts",
+    "prod-migration:schema-drop": "NODE_ENV=production ts-node ./node_modules/typeorm/cli.js schema:drop -d ./dist/config/datasource.js",
```
Comment (MHassanTariq) [inline package.json]: Can we filter out to keep only the ones we will be needing? For instance, do we need to drop the entire schema at any point? 

Plus, I think making prod-migration and local-migration is clustering the scripts in package.json. What if we use a `.env` variable instead to specify our deployment?

File: src/config/ormconfig.ts [hunk 4]
```diff
@@ -1,8 +1,4 @@
 import { DataSourceOptions } from 'typeorm';
-import { config } from 'dotenv';
-
-// Load environment variables from a .env file if necessary
-config();
```
Comment (MHassanTariq) [inline src/config/ormconfig.ts:5]: Removed by mistake? 
We will be needing env variables. No?

**Review by MHassanTariq — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (MHA10) [reply README.md]: aaah yes! Not a npm lover so 😛

File: README.md (hunk 2 above)
Comment (MHA10) [reply README.md]: Yup, it works like a charm.

File: package.json (hunk 3 above)
Comment (MHA10) [reply package.json]: Yup, we can improve this part here.

File: src/config/ormconfig.ts (hunk 4 above)
Comment (MHA10) [reply src/config/ormconfig.ts:5]: added Nest config on app level, so no need of it anymore.

**Comment by MHA10** [comment]
@MHassanTariq 
I'll improve and clean the package.json commands, then you can review those.

**Review by MHassanTariq — APPROVED** [review]

**Pull request merged**